### PR TITLE
The real fastmos: C++ monstermos port

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

A port of https://github.com/yogstation13/Yogstation/pull/7981. I only ported the gas mixture bits, monstermos itself is not ported. This is still LINDA and everything.

I apologize to the maintainers in advance for this.

## Why It's Good For The Game

Fast is good.

## Changelog
:cl:
refactor: Gas mixtures now live entirely in a DLL.
/:cl:

### CODE REVIEW

**Comment by MrJWhit** [comment]
It's _heeereeeeeeee_

File: code/modules/mapping/space_management/multiz_helpers.dm [hunk 1]
```diff
@@ -38,10 +52,11 @@
 	if((holder == NONE) || (holder == (UP|DOWN)))
```
Comment (silicons) [inline code/modules/mapping/space_management/multiz_helpers.dm]:
```
/proc/dir_inverse_multiz(dir)
	var/vertical = dir & (UP|DOWN)
	dir &= ~vertical	// remove dir from vertical
	if(vertical == (UP|DOWN))		// both up/down specified
		vertical = NONE
	if(dir)		// byond will be a shit if dir is 0 so let's not have dir be 0.
		dir = turn(dir, 180)
	return dir | vertical
```
check with monster though, i've already fucked this up once ;)

**Comment by Detective-Google** [comment]
this currently makes snow maps nigh unplayable- we also had a pyro anomaly go off and the medbay got to about 1000K, at which point no amount of heaters and *literally RCDing out the floor* would cool it down

**Comment by silicons** [comment]
this is awesome and @monster860 is based

**Comment by monster860** [comment]
based on what?

**Comment by YakumoChen** [comment]
space wind has been a real bitch to deal with so far.

Also it's kind of broken saunas since water vapor evaporates way too fast now.

**Comment by Putnam3145** [comment]
water vapor evaporation has not been touched in any regard

**Pull request merged**